standalone-metastore/metastore-server/src/main/sql/hive/upgrade-4.0.0-alpha-2-to-4.0.0-beta-1.hive.sql

```diff
@@ -210,6 +210,11 @@ SELECT
   CQ_TBLPROPERTIES
 FROM COMPACTION_QUEUE;
 
+DROP TABLE IF EXISTS `VERSION`;
```

Comment:
why do we need drop if we do create or replace?

Reply:
Thanks. Looks like, we needed it on `upgrade-3.0.0-to-3.1.0` because it was originally created as a physical table.

```
$ git grep '`VERSION`'
hive-schema-3.0.0.hive.sql:CREATE EXTERNAL TABLE IF NOT EXISTS `VERSION` (
hive-schema-3.0.0.hive.sql:INSERT INTO `VERSION` VALUES (1, '3.0.0', 'Hive release version 3.0.0');
hive-schema-3.1.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '3.1.0' AS `SCHEMA_VERSION`,
hive-schema-4.0.0-alpha-1.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-alpha-1' AS `SCHEMA_VERSION`,
hive-schema-4.0.0-alpha-2.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-alpha-2' AS `SCHEMA_VERSION`,
hive-schema-4.0.0-beta-1.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-beta-1' AS `SCHEMA_VERSION`,
hive-schema-4.0.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0' AS `SCHEMA_VERSION`,
hive-schema-4.1.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.1.0' AS `SCHEMA_VERSION`,
upgrade-3.0.0-to-3.1.0.hive.sql:DROP TABLE IF EXISTS `VERSION`;
upgrade-3.0.0-to-3.1.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '3.1.0' AS `SCHEMA_VERSION`,
upgrade-3.1.0-to-4.0.0-alpha-1.hive.sql:DROP TABLE IF EXISTS `VERSION`;
upgrade-3.1.0-to-4.0.0-alpha-1.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-alpha-1' AS `SCHEMA_VERSION`,
upgrade-4.0.0-alpha-1-to-4.0.0-alpha-2.hive.sql:DROP TABLE IF EXISTS `VERSION`;
upgrade-4.0.0-alpha-1-to-4.0.0-alpha-2.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-alpha-2' AS `SCHEMA_VERSION`,
upgrade-4.0.0-alpha-2-to-4.0.0-beta-1.hive.sql:DROP TABLE IF EXISTS `VERSION`;
upgrade-4.0.0-alpha-2-to-4.0.0-beta-1.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-beta-1' AS `SCHEMA_VERSION`,
upgrade-4.0.0-beta-1-to-4.0.0.hive.sql:DROP TABLE IF EXISTS `VERSION`;
upgrade-4.0.0-beta-1-to-4.0.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0' AS `SCHEMA_VERSION`,
upgrade-4.0.0-to-4.1.0.hive.sql:DROP TABLE IF EXISTS `VERSION`;
upgrade-4.0.0-to-4.1.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.1.0' AS `SCHEMA_VERSION`,
```

I updated it so as to minimize DDLs, assuming the upgrade files should be applied one by one in the order.
https://github.com/apache/hive/pull/5324/commits/6478224dd7109a973170b233d737a4b905dc46b3

```
git grep '`VERSION`'             
hive-schema-3.0.0.hive.sql:CREATE EXTERNAL TABLE IF NOT EXISTS `VERSION` (
hive-schema-3.0.0.hive.sql:INSERT INTO `VERSION` VALUES (1, '3.0.0', 'Hive release version 3.0.0');
hive-schema-3.1.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '3.1.0' AS `SCHEMA_VERSION`,
hive-schema-4.0.0-alpha-1.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-alpha-1' AS `SCHEMA_VERSION`,
hive-schema-4.0.0-alpha-2.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-alpha-2' AS `SCHEMA_VERSION`,
hive-schema-4.0.0-beta-1.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-beta-1' AS `SCHEMA_VERSION`,
hive-schema-4.0.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0' AS `SCHEMA_VERSION`,
hive-schema-4.1.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.1.0' AS `SCHEMA_VERSION`,
upgrade-3.0.0-to-3.1.0.hive.sql:DROP TABLE IF EXISTS `VERSION`;
upgrade-3.0.0-to-3.1.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '3.1.0' AS `SCHEMA_VERSION`,
upgrade-3.1.0-to-4.0.0-alpha-1.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-alpha-1' AS `SCHEMA_VERSION`,
upgrade-4.0.0-alpha-1-to-4.0.0-alpha-2.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-alpha-2' AS `SCHEMA_VERSION`,
upgrade-4.0.0-alpha-2-to-4.0.0-beta-1.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0-beta-1' AS `SCHEMA_VERSION`,
upgrade-4.0.0-beta-1-to-4.0.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.0.0' AS `SCHEMA_VERSION`,
upgrade-4.0.0-to-4.1.0.hive.sql:CREATE OR REPLACE VIEW `VERSION` AS SELECT 1 AS `VER_ID`, '4.1.0' AS `SCHEMA_VERSION`,
```

Reply:
👍